ambry-rest/src/main/java/com.github.ambry.rest/StorageServerNettyFactory.java

```diff
@@ -0,0 +1,88 @@
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.socket.SocketChannel;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Netty specific implementation of {@link NioServerFactory}.
+ * <p/>
+ * Sets up all the supporting cast required for the operation of {@link NettyServer} and returns a new instance on
+ * {@link #getNioServer()}.
+ */
+public class StorageServerNettyFactory implements NioServerFactory {
+  private static final Logger LOGGER = LoggerFactory.getLogger(StorageServerNettyFactory.class);
+
+  private final NettyConfig nettyConfig;
+  private final PerformanceConfig performanceConfig;
+  private final NettyMetrics nettyMetrics;
+  final Map<Integer, ChannelInitializer<SocketChannel>> channelInitializers;
+
+  /**
+   * Creates a new instance of NettyFrontendServerFactory.
+   * @param http2Port the port for HTTP2 request.
+   * @param verifiableProperties the in-memory {@link VerifiableProperties} to use.
+   * @param metricRegistry the {@link MetricRegistry} to use.
+   * @param requestHandler the {@link RestRequestHandler} to hand off the requests to.
+   * @param publicAccessLogger the {@link PublicAccessLogger} that can be used for public access logging
+   * @param restServerState the {@link RestServerState} that can be used to check the health of the system
+   *                              to respond to health check requests
+   * @param defaultSslFactory the {@link SSLFactory} used to construct the {@link javax.net.ssl.SSLEngine} used for
+   *                          handling SSL requests (unless {@link NettyConfig#SSL_FACTORY_KEY} is set, in which case
+   *                          it will be overridden).
+   * @throws IllegalArgumentException if any of the arguments are null.
+   * @throws ReflectiveOperationException if a netty-specific {@link SSLFactory} cannot be instantiated via reflection.
+   */
+  public StorageServerNettyFactory(int http2Port, VerifiableProperties verifiableProperties,
+      MetricRegistry metricRegistry, final RestRequestHandler requestHandler,
+      final PublicAccessLogger publicAccessLogger, final RestServerState restServerState, SSLFactory defaultSslFactory)
+      throws ReflectiveOperationException {
+    if (verifiableProperties == null || metricRegistry == null || requestHandler == null || publicAccessLogger == null
+        || restServerState == null) {
+      throw new IllegalArgumentException("Null arg(s) received during instantiation of StorageServerNettyFactory");
+    }
+    nettyConfig = new NettyConfig(verifiableProperties);
+    performanceConfig = new PerformanceConfig(verifiableProperties);
+    nettyMetrics = new NettyMetrics(metricRegistry);
+    ConnectionStatsHandler connectionStatsHandler = new ConnectionStatsHandler(nettyMetrics);
+
+    Map<Integer, ChannelInitializer<SocketChannel>> initializers = new HashMap<>();
```

Comment:
you can just use `Collections.singletonMap` here.

Reply:
cool.